### PR TITLE
API-5357: Manage MFA implementation

### PR DESCRIPTION
**Pull request by syed-ali-hmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatform/modules/mfa/connectors/ThirdPartyDeveloperMfaConnector.scala [hunk 1]
```diff
@@ -77,16 +77,14 @@ class ThirdPartyDeveloperMfaConnector @Inject()(http: HttpClient, config: Applic
     }
   }
 
-  def removeMfa(userId: UserId, email: String)(implicit hc: HeaderCarrier): Future[Unit] = {
-    implicit val RemoveMfaRequestFormat = Json.format[RemoveMfaRequest]
+  def removeMfaById(userId: UserId, mfaId: MfaId, email: String)(implicit hc: HeaderCarrier): Future[Unit] = {
+    implicit val removeMfaByIdRequestFormat = Json.format[RemoveMfaByIdRequest]
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/mfa/connectors/ThirdPartyDeveloperMfaConnector.scala:81]: Put this in the companion of the case class (you can nest object within an object)

File: app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala [hunk 2]
```diff
@@ -38,35 +39,33 @@ import views.html.protectaccount._
...
-  val cookieSigner : CookieSigner,
-  protectAccountSetupView: ProtectAccountSetupView,
-  protectedAccountView: ProtectedAccountView,
-  protectAccountView: ProtectAccountView,
-  protectAccountAccessCodeView: ProtectAccountAccessCodeView,
-  protectAccountCompletedView: ProtectAccountCompletedView,
-  protectAccountRemovalConfirmationView: ProtectAccountRemovalConfirmationView,
-  protectAccountRemovalAccessCodeView: ProtectAccountRemovalAccessCodeView,
-  protectAccountRemovalCompleteView: ProtectAccountRemovalCompleteView
+                                val thirdPartyDeveloperConnector: ThirdPartyDeveloperConnector,
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala:46]: Looks like formatting changes that should not be here.  I don't believe the formatter does this style so probably an IDE issue.

File: app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala [hunk 3]
```diff
@@ -38,35 +39,33 @@ import views.html.protectaccount._
...
+                                protectedAccountMfaRemovalByIdAccessCodeView: ProtectedAccountMfaRemovalByIdAccessCodeView,
+                                protectedAccountRemovalCompleteView: ProtectedAccountRemovalCompleteView
 )(
   implicit val ec: ExecutionContext,
   val appConfig: ApplicationConfig
 ) extends LoggedInController(mcc) with WithDefaultFormBinding {
 
   private val scale = 4
-  val qrCode = QRCode(scale)
+  val qrCode: QRCode = QRCode(scale)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala:68]: For something this obviously typed I would not add the type here.  It's a simple case class and on the same line.  Something that IntelliJ likes to do.  Tell it not to ;-)

File: app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala (hunk 2 above)
Comment (johnsgp) [reply app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala]: Done

File: app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala (hunk 3 above)
Comment (johnsgp) [reply app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/ProtectAccount.scala]: Done

File: app/uk/gov/hmrc/apiplatform/modules/mfa/connectors/ThirdPartyDeveloperMfaConnector.scala (hunk 1 above)
Comment (johnsgp) [reply app/uk/gov/hmrc/apiplatform/modules/mfa/connectors/ThirdPartyDeveloperMfaConnector.scala]: Done

**Review by johnsgp — APPROVED** [review]

**Pull request merged**